graylog2-server/src/main/java/org/graylog2/migrations/V20161216123500_DefaultIndexSetMigration.java

```diff
@@ -0,0 +1,91 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.migrations;
+
+import org.graylog2.configuration.ElasticsearchConfiguration;
+import org.graylog2.events.ClusterEventBus;
+import org.graylog2.indexer.indexset.DefaultIndexSetCreated;
+import org.graylog2.indexer.indexset.IndexSetConfig;
+import org.graylog2.indexer.indexset.IndexSetService;
+import org.graylog2.indexer.indexset.V20161216123500_Succeeded;
+import org.graylog2.indexer.indexset.events.IndexSetCreatedEvent;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Migration for moving indexing settings into existing index sets.
+ */
+public class V20161216123500_DefaultIndexSetMigration extends Migration {
+    private static final Logger LOG = LoggerFactory.getLogger(V20161216123500_DefaultIndexSetMigration.class);
+
+    private final ElasticsearchConfiguration elasticsearchConfiguration;
+    private final IndexSetService indexSetService;
+    private final ClusterConfigService clusterConfigService;
+    private final ClusterEventBus clusterEventBus;
+
+    @Inject
+    public V20161216123500_DefaultIndexSetMigration(final ElasticsearchConfiguration elasticsearchConfiguration,
+                                                    final IndexSetService indexSetService,
+                                                    final ClusterConfigService clusterConfigService,
+                                                    final ClusterEventBus clusterEventBus) {
+        this.elasticsearchConfiguration = elasticsearchConfiguration;
+        this.indexSetService = indexSetService;
+        this.clusterConfigService = clusterConfigService;
+        this.clusterEventBus = clusterEventBus;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.of(2016, 12, 16, 12, 35, 0, 0, ZoneOffset.UTC);
+    }
+
+    @Override
+    public void upgrade() {
+        if (clusterConfigService.get(V20161216123500_Succeeded.class) != null) {
+            return;
+        }
+
+        // The default index set must have been created first.
+        checkState(clusterConfigService.get(DefaultIndexSetCreated.class) != null, "The default index set hasn't been created yet. This is a bug!");
+
+        for (IndexSetConfig indexSetConfig : indexSetService.findAll()) {
+            final String analyzer = elasticsearchConfiguration.getAnalyzer();
+            final IndexSetConfig updatedConfig = indexSetConfig.toBuilder()
+                    .indexAnalyzer(analyzer)
+                    .indexTemplateName(elasticsearchConfiguration.getTemplateName())
```

Comment:
Because otherwise we will overwrite the same index template for every index set. Every index set has a different wildcard `template` field.

Example:

```
http :9200/_template | jq '.[].template'
"emails_*"
"yolo_*"
"restored-archive-*"
"graylog_*"
```